core/tx_pool.go

```diff
@@ -291,19 +307,24 @@ func (self *TxPool) add(tx *types.Transaction) error {
 	}
 	self.queueTx(hash, tx)
 
-	if glog.V(logger.Debug) {
-		var toname string
-		if to := tx.To(); to != nil {
-			toname = common.Bytes2Hex(to[:4])
-		} else {
-			toname = "[NEW_CONTRACT]"
-		}
-		// we can ignore the error here because From is
-		// verified in ValidateTransaction.
-		f, _ := types.Sender(self.signer, tx)
-		from := common.Bytes2Hex(f[:4])
-		glog.Infof("(t) %x => %s (%v) %x\n", from, toname, tx.Value, hash)
-	}
+	var toname string
+	if to := tx.To(); to != nil {
+		toname = common.Bytes2Hex(to[:4])
+	} else {
+		toname = "[NEW_CONTRACT]"
+	}
+	// we can ignore the error here because From is
+	// verified in ValidateTransaction.
+	f, _ := types.Sender(self.signer, tx)
+	from := common.Bytes2Hex(f[:4])
+
+	mlogTxPool.Send(mlogTxPoolAddTx.SetDetailValues(
```

Comment:
There's a command line flag `mlog=off` to disable logging. Should we avoid sending anything to mlog in that case? Just might help to keep things fast.

Reply:
Roger that -- I'm about to push changes to implement avoiding sending anything to mlog ✅  

To be honest.. I'm not sure what the performance saving/penalty is because `<mlogger>.Send()` checks the registry for the <mlogger>'s existence... 1.) I think you are probably right that it will be faster to check a boolean than get a read-lock on a map and check for key existence, and 2.) not sure (and curious) if arguments to `<mlogger>.Send(these, arguments)` are evaluated "on-demand" (as/if `Send` needs them) or regardless of whether Send will wind up using them...